writer_downgrade_test.go

```diff
@@ -77,10 +77,15 @@ func (s *WriterDowngradeSuite) TestWriteObjectAndDowngrade() {
 		Trust    bool   `rsf:"trust"` // Not in original struct
 	}
 	// Add a new type that was also not included in the original data.
+	type variation struct {
+		Id          int8   `rsf:"id,skip"`
+		Description string `rsf:"description"`
+	}
 	type product struct {
-		Barcode string  `rsf:"barcode,skip,fixed:12"`
-		Name    string  `rsf:"name"`
-		Price   float32 `rsf:"price"`
+		Barcode    string      `rsf:"barcode,skip,fixed:12"`
+		Name       string      `rsf:"name"`
+		Price      float32     `rsf:"price"`
+		Variations []variation `rsf:"variations,index:id"`
```

Comment:
What happens if you set `index` to a field that doesn't exist?

Reply:
I'm not sure! I was actually thinking that we should add some error handling around this case, but it wasn't directly related to what I was trying to do in this PR.

Reply:
No need for this PR then, mostly just curious how it would behave!